superset-frontend/packages/superset-ui-chart-controls/src/sections/timeComparison.tsx

```diff
@@ -94,9 +98,29 @@ export const timeComparisonControls: ({
         name: 'start_date_offset',
         config: {
           type: 'TimeOffsetControl',
-          label: t('shift start date'),
+          label: t('Shift start date'),
           visibility: ({ controls }) =>
             controls?.time_compare.value === 'custom',
+          mapStateToProps: (
+            state: ControlPanelState,
+            controlState: ControlState,
+          ) => {
+            const { form_data } = state;
+            const { time_compare } = form_data;
+            const newState = { ...controlState };
+            if (
+              time_compare === 'custom' &&
+              (controlState.value === '' ||
+                controlState.value === 'Invalid date')
+            ) {
+              newState.externalValidationErrors = [
+                t('A date is required when using Custom date shift'),
```

Comment:
Is this "Custom" with capital C intended?

Reply:
Oh! Good catch, thanks